cli/test/request-id-interceptor.test.ts

```diff
@@ -0,0 +1,89 @@
+import { afterEach, describe, test, expect, vi } from 'vitest';
+import { EnumStatusCode } from '@wundergraph/cosmo-connect/dist/common/common_pb';
+import { requestIdInterceptor } from '../src/core/client/client.js';
+
+let mockedId = 'mocked-id';
+
+const consoleLogSpy = vi.spyOn(console, 'log').mockImplementation(() => {});
+
+vi.mock('node:crypto', () => ({
+  randomUUID: () => mockedId,
+}));
+
+const ERR_RESPONSE = { message: { response: { code: EnumStatusCode.ERR, details: 'An error occurred' } }, method: { name: 'WhoAmI' } };
+const SUCCESS_RESPONSE = { message: { response: { code: EnumStatusCode.OK, details: 'No error here' } } };
+
+
+describe('requestIdInterceptor()', () => {
```

Comment:
Instead of mocking it you could instantiate a buf client in the tests and provide a dummy service implementation that validates end to end if the request ID arrives. In the same way, you could also test that in case of an error the informations are logged.